examples/post-item-with-pystac.ipynb

```diff
@@ -0,0 +1,290 @@
+    "2020-04-15 15:23:30  324896818 berlin-catalog/image/berlin-not-cog.tif\n",
+    "2020-04-14 12:59:43       1553 berlin-catalog/image/berlin-sentinel.json\n",
+    "2020-04-15 15:23:30  112040499 berlin-catalog/image/berlin.tif\n",
+    "2020-04-14 10:50:46     587090 berlin-catalog/labels/berlin-labels.tif\n",
+    "2020-04-23 17:17:17       1880 berlin-catalog/labels/bldg-labels.json\n",
+    "2020-04-15 15:24:04    9005460 berlin-catalog/labels/colormap.png\n",
+    "2020-04-15 15:24:06    9005460 berlin-catalog/labels/colormap.tif\n",
+    "2020-04-14 10:50:48     666253 berlin-catalog/predictions/berlin-preds.tif\n",
+    "2020-04-24 15:38:20       2160 berlin-catalog/predictions/bldg-predictions.json\n",
+    "2020-04-24 15:36:54       5367 berlin-catalog/predictions/some-random-boxes.json\n",
+    "```\n",
+    "\n",
+    "We'll read the `berlin-sentinel.json` item with a custom `read_text_method` in `STAC_IO` that's aware of how to read from S3.\n",
+    "\n",
+    "**Note**: if you don't have a `default` AWS profile, you'll need to set one up using `aws configure` from the [AWS CLI](https://aws.amazon.com/cli/)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(<Item id=image>, <Catalog id=berlin>)"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import boto3\n",
+    "from pystac import Collection, Item\n",
+    "from pystac.stac_io import STAC_IO\n",
+    "from urllib.parse import urlparse\n",
+    "\n",
+    "s3_client = boto3.client('s3')\n",
+    "\n",
+    "def read_text_s3(uri: str) -> str:\n",
+    "    parsed = urlparse(uri)\n",
+    "    bucket = parsed.hostname\n",
+    "    key = parsed.path.lstrip(\"/\")\n",
+    "    return s3_client.get_object(Bucket=bucket, Key=key)[\"Body\"].read()\n",
+    "\n",
+    "STAC_IO.read_text_method = read_text_s3\n",
+    "\n",
+    "item = Item.from_file(\"s3://rasterfoundry-development-data-us-east-1/berlin-catalog/image/berlin-sentinel.json\")\n",
+    "(item, item.get_parent())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Find collections in Franklin\n",
+    "\n",
+    "Franklin advertises its collections under the `/collections` endpoint. For this example we'll use an existing collection called `berlin` to POST items to."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[<Collection id=berlin>]"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import requests\n",
+    "\n",
+    "[Collection.from_dict(x) for x in requests.get(\"http://localhost:9090/collections\").json()[\"collections\"]]"
```

Comment:
I think it would be good to parameterize with this at a variable at the top for demonstration purposes - this way we can demo on a remote instance if we'd like